src/components/pokemon-list/PokemonList.tsx

```diff
@@ -30,23 +31,35 @@ const preloadBackImages = () => {
 }
 
 const PokemonList = ({searchTerm} :PokemonListProps) => {
+    const[limit,setLimit]=useState(9)
+    const debouncedLimit = useDebounce(limit, 300);
     useEffect(() => {
-        preloadBackImages();
+        preloadBackImages(debouncedLimit);
     }, []);
 
     const cards = pokemonList
-        .slice(0, limit)
         .filter(({name}) => name.toLowerCase().includes(searchTerm?.toLowerCase()))
+        .slice(0, debouncedLimit)
         .map(pokemon => {
             return <PokemonCard
                 key={pokemon.id}
                 pokemon={pokemon}
             />;
     });
 
-    return <div className="pokemon-list">
+    return <>
+    <input type="range" 
+    min="0" 
+    max="1000" 
+    value={limit} 
+    className="range" 
+    onChange={e => setLimit(parseInt(e.target.value))} />
```

Comment:
the range input has a size of the visible screen, should be better to make it smaller, like  1/3rd  of the screen